src/components/submitButton.jsx

```diff
@@ -1,5 +1,10 @@
+// @flow
 import React from 'react';
 
-const SubmitButton = ({ action, values }) => <button onClick={(e) => action(e, values)}>Submit</button>;
+// types
+import type { Element } from 'react';
+import type { Values } from '../lib/type';
+
+const SubmitButton = ({ action, values }: { action: Function, values: Values}): Element<*> => <button onClick={(e: {preventDefault: Function}) => action(e, values)}>Submit</button>;
```

Comment:
The 'e' type is great. However, you use this later on. Inline typing is a quick resolve for flow. As you look at the application as a whole, where else is this being used? Can this be combined so that there is an export type used in multiple locations or is this really a single, one time, use?